packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,182 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJson } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunOnBrowser = !fs.readFileSync;
+
+export type SvelteContext = {
+	svelteVersion: string;
+	fileType: '.svelte' | '.svelte.[js|ts]';
+	runes: boolean;
+	svelteKitVersion: string | null;
+	svelteKitFileType:
+		| '+page.svelte'
+		| '+page.js'
+		| '+page.server.js'
+		| '+error.svelte'
+		| '+layout.svelte'
+		| '+layout.js'
+		| '+layout.server.js'
+		| '+server.js'
+		| null;
+};
+
+function getFileType(filePath: string): SvelteContext['fileType'] | null {
+	if (filePath.endsWith('.svelte')) {
+		return '.svelte';
+	}
+
+	if (filePath.endsWith('.svelte.js') || filePath.endsWith('.svelte.ts')) {
+		return '.svelte.[js|ts]';
+	}
+
+	return null;
+}
+
+function getSvelteKitFileTypeFromFilePath(filePath: string): SvelteContext['svelteKitFileType'] {
+	const fileName = filePath.split('/').pop();
+	switch (fileName) {
+		case '+page.svelte': {
+			return '+page.svelte';
+		}
+		case '+page.js':
+		case '+page.ts': {
+			return '+page.js';
+		}
+		case '+page.server.js':
+		case '+page.server.ts': {
+			return '+page.server.js';
+		}
+		case '+error.svelte': {
+			return '+error.svelte';
+		}
+		case '+layout.svelte': {
+			return '+layout.svelte';
+		}
+		case '+layout.js':
+		case '+layout.ts': {
+			return '+layout.js';
+		}
+		case '+layout.server.js':
+		case '+layout.server.ts': {
+			return '+layout.server.js';
+		}
+		case '+server.js':
+		case '+server.ts': {
+			return '+server.js';
+		}
+		default: {
+			return null;
+		}
+	}
+}
+
+function getSvelteKitContext(
+	context: RuleContext
+): Pick<SvelteContext, 'svelteKitFileType' | 'svelteKitVersion'> {
+	const filePath = getFilename(context);
+	const svelteKitVersion = gteSvelteKitVersion(filePath);
+	if (svelteKitVersion == null) {
+		return {
+			svelteKitFileType: null,
+			svelteKitVersion: null
+		};
+	}
+	if (isRunOnBrowser) {
+		return {
+			svelteKitVersion,
+			// Judge by only file path if it runs on browser.
+			svelteKitFileType: getSvelteKitFileTypeFromFilePath(filePath)
+		};
+	}
+
+	const routes =
+		(
+			context.settings?.svelte?.kit?.files?.routes ??
+			getSourceCode(context).parserServices.svelteParseContext?.svelteConfig?.kit?.files?.routes
+		)?.replace(/^\//, '') ?? 'src/routes';
+	const projectRootDir = getProjectRootDir(getFilename(context)) ?? '';
+
+	if (!filePath.startsWith(path.join(projectRootDir, routes))) {
+		return {
+			svelteKitVersion,
+			svelteKitFileType: null
+		};
+	}
+
+	return {
+		svelteKitVersion,
+		svelteKitFileType: getSvelteKitFileTypeFromFilePath(filePath)
+	};
+}
+
+/**
+ * Check givin file is under SvelteKit project.
+ *
+ * If it runs on browser, it always returns true.
+ *
+ * @param filePath A file path.
+ * @returns
+ */
+function gteSvelteKitVersion(filePath: string): string | null {
```

Comment:
Typo (`gte` instead of `get`)

Reply:
updated: https://github.com/sveltejs/eslint-plugin-svelte/pull/980/commits/910effd2057a67921a6ed0b01be5de31aae52d4a